tools/mcp/dotnet/AzureSDKMCPHub/Azure.SDK.Tools.MCP.Hub/Tools/README.md

```diff
@@ -0,0 +1,21 @@
+# Writing tools that integrate with the MCP server
+
+## How to create your tool code
+
+- Inherit from `Azure.SDK.Tools.MCP.Contract::MCPHubTool`
+- Ensure your class has the `[McpServerToolType]` attribute added to it
+  - Eventual goal is to automatically recognize any classes that have this type as `[McpServerToolType]`.
+  - For now you need to add the attribute yourself.
+- Mark each function with `[McpServerTool]`.
+- Leverage `HelloWorld.cs` for a functioning basic example (more advanced samples to follow)
+- If you require any helper functions or other shared code. Place your code under a _folder_ named the same as your Tool class name.
+  - See `ReleasePlanTool` for an example of this folder structure.  
+
+## To quickly dev loop on just your class.
+
+- Ensure that your class in `Tools` is inheriting from the abstract class `MCPHubTool.`
+- Update the `debug` properties for the `hub` project to target a specific tool `--tool <nameofyourclass>`
```

Comment:
I think long term we'll need some official way of building or running the server with only a set of tools and/or functions available (e.g. exposing the rest-api-spec MCP to partners without also having things like `eng/common PR checker` exposed.

Reply:
Understood. With external packages that's simpler. With the internal classes -- we'll have to think a bit.